### PR TITLE
feat: add gRPC listener for new write protocol

### PR DESCRIPTION
**Pull request by jacobmarble**
Creates a new write service using gRPC/protobuf.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by mkmik — APPROVED** [review]
looks good!

File: generated_types/protos/influxdata/transfer/column/v1/write.proto [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  // Mask that maps the position to if that value is null. Null positions will
+  // not have a value represented in the values array. To read the values out
+  // of the column requires traversing the mask to determine what position in
+  // the values array that index is located in. Here's what it might look like:
+  //    position:   7      0  15     8
+  //         bit:   00100011  00111000
+  // TODO note about endianness
+  // An on bit (1) indicates that the value at that position is null. If there are
+  // no null values in the column, the null_mask is omitted from the flatbuffers.
+  // TODO test repeated bool instead of this
```
Comment (crepererum) [inline generated_types/protos/influxdata/transfer/column/v1/write.proto:70]: is this still a TODO or has the format settled on bytes?

Comment (mkmik) [reply generated_types/protos/influxdata/transfer/column/v1/write.proto:70]: btw, my assumption was that this file is not the canonical copy of the the "new ingestion protocol". Is there a canonical place where this protocol definition lives? Can we link to it?

**Comment by alamb** [comment]
I wonder if it would be worth adding an end to end test for this client library to validate the plumbing is setup correctly.

Perhaps something in the vein of https://github.com/influxdata/influxdb_iox/blob/main/tests/end_to_end_cases/flight_api.rs

File: generated_types/protos/influxdata/transfer/column/v1/write.proto (hunk 1 above)
Comment (jacobmarble) [reply generated_types/protos/influxdata/transfer/column/v1/write.proto:70]: Sorry, I took the earlier approval and ran with it.

The format for the null mask is bytes. I'll link to the canonical location once it exists. Upper management have not given conclusive feedback on the final name yet, so I can't create an official repository.

**Pull request merged**